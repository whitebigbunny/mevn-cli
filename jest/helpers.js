'use strict';

import execa from 'execa';
import path from 'path';

const CLI_PATH = path.resolve(process.cwd(), 'bin', 'mevn.js');

export const run = args => execa.sync(CLI_PATH, args);
