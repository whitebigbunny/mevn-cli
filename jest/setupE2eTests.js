jest.setTimeout(350000);
