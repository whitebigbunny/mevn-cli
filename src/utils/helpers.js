const fs = require('fs');

const { isWin } = require('./constants');

/**
 * Strips \r character from the file content if the host OS is windows
 *
 * @param {String[]} fileContent - The file content
 * @returns {String[]}
 */

const stripChar = (fileContent) =>
  fileContent.map((content) =>
    content.includes('\r') ? content.substr(0, content.indexOf('\r')) : content,
  );
/**
 * Returns the file content as an arrray
 *
 * @param {String} filePath - The file path
 * @returns {String[]}
 */

const readFileContent = (filePath) => {
  const fileContent = fs.readFileSync(filePath, 'utf8').split('\n');

  // Check if the host OS is windows
  if (isWin) {
    return stripChar(fileContent);
  }
  return fileContent;
};

module.exports = readFileContent;
