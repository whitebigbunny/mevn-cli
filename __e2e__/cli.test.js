'use strict';

import { run } from '../jest/helpers';
// Generic tests

test('shows up help if no arguments were passed', () => {
  const { stdout } = run();
  expect(stdout).toMatchSnapshot();
});

test('show up help information on passing in the respective options', () => {
  ['-h', '--help'].forEach(op => {
    const { stdout } = run([op]);
    expect(stdout).toMatchSnapshot();
  });
});

test('show up CLI version information', () => {
  ['-V', '--version'].forEach(op => {
    const { stdout } = run([op]);
    expect(stdout).toMatchSnapshot();
  });
});

test('warns the user on passing in unknown option', () => {
  try {
    const { stderr } = run(['--invalid']);
    expect(stderr).to.be(`error: unknown option '--invalid'`);
  } catch (err) {
    // handle err
  }
});

// A bit specific

test('warns the user if an unknown command is passed', () => {
  const { stdout } = run(['junkcmd']);
  expect(stdout).toMatchSnapshot();
});

test('suggests the matching command if the user makes a typo', () => {
  const { stdout } = run(['ini']);
  expect(stdout).toMatchSnapshot();
});
