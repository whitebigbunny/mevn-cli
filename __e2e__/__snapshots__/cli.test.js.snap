// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show up CLI version information 1`] = `"1.5.1"`;

exports[`show up CLI version information 2`] = `"1.5.1"`;

exports[`show up help information on passing in the respective options 1`] = `
"
  Usage: mevn <command> [options]

  Options:

    -V, --version              output the version number
    -h, --help                 output usage information

  Commands:

    init <appname>             To init the project
    codesplit <componentname>  To code split the required component
    generate                   To generate model, route, controller and DB config files
    add:package                To add a new package to the project
    serve                      To serve client/server
    dockerize                  To dockerize the app
    deploy                     To deploy the app to Heroku
    info                       Prints debugging information about the local environment"
`;

exports[`show up help information on passing in the respective options 2`] = `
"
  Usage: mevn <command> [options]

  Options:

    -V, --version              output the version number
    -h, --help                 output usage information

  Commands:

    init <appname>             To init the project
    codesplit <componentname>  To code split the required component
    generate                   To generate model, route, controller and DB config files
    add:package                To add a new package to the project
    serve                      To serve client/server
    dockerize                  To dockerize the app
    deploy                     To deploy the app to Heroku
    info                       Prints debugging information about the local environment"
`;

exports[`shows up help if no arguments were passed 1`] = `
"
  Usage: mevn <command> [options]

  Options:

    -V, --version              output the version number
    -h, --help                 output usage information

  Commands:

    init <appname>             To init the project
    codesplit <componentname>  To code split the required component
    generate                   To generate model, route, controller and DB config files
    add:package                To add a new package to the project
    serve                      To serve client/server
    dockerize                  To dockerize the app
    deploy                     To deploy the app to Heroku
    info                       Prints debugging information about the local environment"
`;

exports[`suggests the matching command if the user makes a typo 1`] = `
"
  Usage: mevn <command> [options]

  Options:

    -V, --version              output the version number
    -h, --help                 output usage information

  Commands:

    init <appname>             To init the project
    codesplit <componentname>  To code split the required component
    generate                   To generate model, route, controller and DB config files
    add:package                To add a new package to the project
    serve                      To serve client/server
    dockerize                  To dockerize the app
    deploy                     To deploy the app to Heroku
    info                       Prints debugging information about the local environment
  
  Unknown command ini.

  Did you mean init?"
`;

exports[`warns the user if an unknown command is passed 1`] = `
"
  Usage: mevn <command> [options]

  Options:

    -V, --version              output the version number
    -h, --help                 output usage information

  Commands:

    init <appname>             To init the project
    codesplit <componentname>  To code split the required component
    generate                   To generate model, route, controller and DB config files
    add:package                To add a new package to the project
    serve                      To serve client/server
    dockerize                  To dockerize the app
    deploy                     To deploy the app to Heroku
    info                       Prints debugging information about the local environment
  
  Unknown command junkcmd.
"
`;
